src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp

```diff
@@ -736,6 +713,61 @@ class TPackageScanner final : public AbstractHandler<std::shared_ptr<TScanContex
                     CNAValue.c_str(),
                     e.what());
         }
+    }
+
+public:
+    // LCOV_EXCL_START
+    /**
+     * @brief PackageScanner constructor.
+     *
+     * @param databaseFeedManager Database feed manager.
+     */
+    explicit TPackageScanner(std::shared_ptr<TDatabaseFeedManager>& databaseFeedManager)
+        : m_databaseFeedManager(databaseFeedManager)
+    {
+    }
+    // LCOV_EXCL_STOP
+
+    /**
+     * @brief Handles request and passes control to the next step of the chain.
+     *
+     * @param data Scan context.
+     * @return std::shared_ptr<TScanContext> Abstract handler.
+     */
+    std::shared_ptr<TScanContext> handleRequest(std::shared_ptr<TScanContext> data) override
+    {
+        data->m_vulnerabilitySource = getCNA(data);
+
+        const auto& CNAValue = data->m_vulnerabilitySource.second;
+
+        if (DEFAULT_CNA != CNAValue)
+        {
+            packageScan(data, CNAValue);
+        }
+        else
+        {
+            // If the default CNA is used, we verify the CNA priority array
+            if (TGlobalData::instance().vendorMaps().contains(ADP_DEFAULT_ARRAY_KEY))
+            {
+                auto& defaultCNAsArray = TGlobalData::instance().vendorMaps().at(ADP_DEFAULT_ARRAY_KEY);
+
+                for (const nlohmann::json& cna : defaultCNAsArray)
+                {
+                    logDebug2(WM_VULNSCAN_LOGTAG, "Using CNA '%s' from CNA array.", cna.get<std::string>().c_str());
+                    packageScan(data, cna);
+                    if (!data->m_elements.empty())
```

Comment:
I think this logic is incorrect, basically what we should do is scan in order but not replace previous detections with the new one.


I give an example, imagine that CISA detects CVE-2025-1234 and CVE-2025-5678, but NVD detects CVE-2025-1234 and CVE-2025-4444,

The result within the m_elements array must be.

CVE-2025-1234 -> with CISA context
CVE-2025-5678 -> with CISA context.
CVE-2025-4444 -> with NVD context.


Therefore, I want to say that it should be scanned with both FEEDs, but the CISA FEED has more priority than the NVD FEED when it comes to giving the result context.